addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/IPBridgeThingHandler.java

```diff
@@ -0,0 +1,119 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.knx.KNXBindingConstants;
+import org.openhab.binding.knx.internal.handler.BridgeConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tuwien.auto.calimero.IndividualAddress;
+import tuwien.auto.calimero.exception.KNXException;
+import tuwien.auto.calimero.link.KNXNetworkLink;
+import tuwien.auto.calimero.link.KNXNetworkLinkIP;
+import tuwien.auto.calimero.link.medium.TPSettings;
+
+/**
+ * The {@link IPBridgeThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels. It implements a KNX/IP Gateway, that either acts a a
+ * conduit for other {@link KNXBasicThingHandler}s, or for Channels that are
+ * directly defined on the bridge
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public class IPBridgeThingHandler extends KNXBridgeBaseThingHandler {
+
+    private static final String MODE_ROUTER = "ROUTER";
+    private static final String MODE_TUNNEL = "TUNNEL";
+
+    private final Logger logger = LoggerFactory.getLogger(IPBridgeThingHandler.class);
+
+    private int ipConnectionType;
+    private String ip;
+    private String localSource;
+    private int port;
+    private InetSocketAddress localEndPoint;
+    private Boolean useNAT;
+
+    public IPBridgeThingHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        localSource = (String) getConfig().get(LOCAL_SOURCE_ADDRESS);
+
+        String connectionTypeString = (String) getConfig().get(IP_CONNECTION_TYPE);
+        if (StringUtils.isNotBlank(connectionTypeString)) {
+            if (MODE_TUNNEL.equalsIgnoreCase(connectionTypeString)) {
+                ip = (String) getConfig().get(IP_ADDRESS);
+                port = ((BigDecimal) getConfig().get(PORT_NUMBER)).intValue();
+                ipConnectionType = KNXNetworkLinkIP.TUNNELING;
+                useNAT = getConfigAs(BridgeConfiguration.class).getUseNAT();
+            } else if (MODE_ROUTER.equalsIgnoreCase(connectionTypeString)) {
+                ipConnectionType = KNXNetworkLinkIP.ROUTING;
+                if (StringUtils.isBlank(ip)) {
+                    ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
+                }
+                if (StringUtils.isBlank(localSource)) {
+                    localSource = KNXBindingConstants.DEFAULT_LOCAL_SOURCE_ADDRESS;
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        MessageFormat.format(
+                                "Unknown IP connection type {0}. Known types are either 'TUNNEL' or 'ROUTER'",
+                                connectionTypeString));
+                return;
+            }
+        } else {
+            ipConnectionType = KNXNetworkLinkIP.TUNNELING;
+        }
+
+        try {
+            if (!useNAT) {
```

Comment:
@sjka NPE if MODE_ROUTER is set. Either have to move line #70 or initialise at line #54